crates/core/src/network/connection_manager.rs

```diff
@@ -386,20 +526,30 @@ impl ConnectionManager {
                 let fut = {
                     let dest = dest.clone();
                     let transport_connector = transport_connector.clone();
-                    let router = self.inner.lock().router.clone();
                     let conn_tracker = self.clone();
-                    let peer_router = self.clone();
                     async move {
+                        // Gossip swimlane is always bidirectional. If a node A connects to node B
+                        // via the gossip swimlane, there is no need to establish a connection in
+                        // the other direction. If this is proving to be problematic, it's okay to
+                        // revert back to using Forward. Such a change will be backwards
+                        // compatible.
+                        let direction = if swimlane == Swimlane::Gossip {
+                            ConnectionDirection::Bidirectional
+                        } else {
+                            ConnectionDirection::Forward
+                        };
```

Comment:
The method name might be a bit misleading with this change since it suggests that we are creating a forward connection.

Reply:
agree, will rename.